PyEMD/checks.py

```diff
@@ -67,49 +75,43 @@ def whitenoise_check(data: np.ndarray, method: str='EMD',  test: str='aposterior
     >>> type(significant_imfs)
     <class 'dict'>
     """
+    assert 0 < alpha < 1, "alpha value should be in between (0,1)"
+    assert test == "apriori" or test == "aposteriori", "Invalid test type"
+    assert isinstance(IMFs, np.ndarray), "Invalid Data type, Pass a numpy.ndarray containing IMFs"
+    assert rescaling_imf > 0 and rescaling_imf <= len(IMFs), "Invalid rescaling IMF"
 
-    assert alpha > 0 and alpha < 1, "alpha value should be in between (0,1)"
-    assert method == 'EMD' or method == 'EEMD' or method == 'CEEMDAN', 'Invalid decomposition method'
-    assert test == 'apriori' or test == 'aposteriori', 'Invalid test type'
-    assert isinstance(data, np.ndarray), 'Invalid Data type, Pass a numpy.ndarray'
+    N = len(IMFs[0])
+    output = {}
+    if N == 0:
+        return {}
+    if np.isnan(np.sum(IMFs)) is True:
+        # Return NaN if input has NaN
+        return None
```

Comment:
From user's perspective it won't be understandable why `None` is returned. It isn't stated in the doc string for the method and there isn't any logging. Maybe updating docstring with this corner case and the code could be something like

```python
# import logging at the top of the file

if np.isnan(np.sum(IMFs)):
    logging.getLogger("PyEMD").warning("Detected NaN values during whitenoise check. Skipping check.")
    return None
```

Reply:
done